server/src/test/java/org/elasticsearch/index/fielddata/SortingNumericDocValuesTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.fielddata;
+
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matchers;
+
+import java.util.function.LongConsumer;
+
+public class SortingNumericDocValuesTests extends ESTestCase {
+
+   public void testResize() {
+       LongConsumer consumer = new LongConsumer() {
+           long total = 0;
+           @Override
+           public void accept(long value) {
+               total += value;
+               assertThat(total, Matchers.greaterThanOrEqualTo(0L));
```

Comment:
I think it'd be ever so slightly more readable to make the `LongConsumer` something like `AtomicLong::getAndAdd` and then assert the size is exactly what you expect at the bottom. I guess it doesn't assert that you don't have transient negative values though. Maybe do both? Not sure. This is fine too, really.